deep_qa/layers/vector_matrix_merge.py

```diff
@@ -67,17 +67,32 @@ def compute_output_shape(self, input_shapes):
     def compute_mask(self, inputs, mask=None):
         if mask is None or all(m is None for m in mask) or not self.propagate_mask:
             return None
-        print(mask)
+        mask_concat_axis = self.mask_concat_axis
+        if mask_concat_axis is None:
+            mask_concat_axis = self.concat_axis
+            if mask_concat_axis < 0:
+                mask_concat_axis %= K.ndim(inputs[-1])
         num_vectors = len(mask) - 1
-        matrix_mask = K.cast(mask[-1], 'uint8')
-        # We're just trying to get a mask of the right shape, here, so we can call K.ones_like() on
-        # it.  We'll ignore the actual values in this.
-        vector_mask_template = K.sum(matrix_mask, axis=self.mask_concat_axis)
-        vector_mask = K.expand_dims(K.ones_like(vector_mask_template), axis=self.mask_concat_axis)
+        matrix_mask = mask[-1]
         result_mask = matrix_mask
-        for _ in range(num_vectors):
-            result_mask = K.concatenate([vector_mask, result_mask], axis=self.mask_concat_axis)
-        return K.cast(result_mask, 'bool')
+        if mask_concat_axis >= K.ndim(matrix_mask):
+            # This means we're concatenating along an axis that is lower than the mask.  E.g.,
```

Comment:
You mean greater here right? (In the sense of further to the right in terms of number of dimensions of the tensor)